woocommerce/woocommerce
# [CYS on Core] Display a new modal informing the theme switch whenever the user clicks on the "Start designing" button

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Ensure a new modal is displayed whenever the store owner clicks on the "Start designing" button, informing their theme will be switched if they proceed.

Closes https://github.com/woocommerce/woocommerce/issues/44820



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure the `WooCommerce Beta Tester` plugin is installed and activated (available on this monorepo).
2. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
3. Go to `WooCommerce > Home` and click on `Start designing`.
4. Make sure the following modal is opened:

<img width="444" alt="Screenshot 2024-02-21 at 11 50 37" src="https://github.com/woocommerce/woocommerce/assets/15730971/6cbd25c9-b715-4dd0-a4fd-5f25d169bfb0">

5. Ensure that whenever you click on Cancel or on the "X" icon, the modal is closed
6. Ensure that whenever you click on Design a new Theme, you are led to the CYS loading screen & the pattern assembler.
7. Now test on WooExpress and make sure nothing has been changed on that environment.

Demo:

https://github.com/woocommerce/woocommerce/assets/15730971/cf8d417e-b378-453b-bcd6-476b17f00ea2



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Display a new modal informing the theme switch whenever the user clicks on the "Start designing" button in the Customize Your Store flow.

#### Comment 

</details>
